R/EDA_understory.R

```diff
@@ -0,0 +1,171 @@
+
+  raw_data <- FilterPACNVeg("Understory", park, sample_frame, community, year, cycle, plot_type, is_qa_plot = FALSE, silent = silent)
+  if (combine_strata == TRUE) {
+    raw_data <- UnderCombineStrata(raw_data)
+  }
+
+  # Calculate Total Native & Nonnative Cover by stratum
+  Nat_Cov <- raw_data %>%
+    #'*Dead Still present in Understory Data*
+    dplyr::filter(Dead==FALSE)  %>%
+    ## Drop point records if point had no hits: (drop if 'Code == NA')
+    tidyr::drop_na(Code)  %>%
+    # Count Species at each cover point (for Native & Non-native within each Strata):
+    dplyr::group_by(Unit_Code, Sampling_Frame, Plot_Type, Plot_Number, Point, Stratum, Nativity, Year, Cycle) %>%
+    dplyr::summarise(Hits_All_Nat = dplyr::n(), .groups = 'drop')  %>%
+    # group hits by plot (remove Point from grouping variable)
+    dplyr::group_by(Unit_Code, Sampling_Frame, Plot_Type, Plot_Number, Stratum, Nativity, Year, Cycle) %>%
+    # Total hits at each point for each strata for entire plot
+    #   (can be > 300 points or >100% because more than one native species can be present per point)
+    dplyr::summarise(tot_pct_cov = (sum(Hits_All_Nat)) / 300 * 100, .groups = 'drop') %>%
+    # Insert "0" for cover if category does not exsist (for example no hits for non-natives in High Stratum)
+    tidyr::complete(tidyr::nesting(Unit_Code, Sampling_Frame, Plot_Type, Plot_Number, Year, Cycle), Stratum, Nativity,
+                    fill = list(tot_pct_cov = 0)) %>%
+    # Arrange table so that difference in cover between cycles can be calculated easily (example - cycle 1 value for
+    #   cover is followed by cycle 2 value for cover).
+    dplyr::group_by(Unit_Code, Sampling_Frame, Plot_Type, Plot_Number, Stratum, Nativity) %>%
+    dplyr::arrange(Cycle, .by_group = TRUE) %>%
+    dplyr::ungroup()
+
+  if (paired_change == TRUE) {
+    paired_plots <- RemoveSingleVisits(raw_data)
+    p <- paired_plots$Plot_Number
+    Nat_Cov <- Nat_Cov %>%
+      # remove plots that were only sampled once (this removes all rotationals and possibly some fixed plots if only sampled once)
+      dplyr::filter(Plot_Number %in% p) %>%
+      dplyr::group_by(Unit_Code, Sampling_Frame, Plot_Type, Plot_Number, Stratum, Nativity) %>%
+      # Calculate the change in cover per cycle
+      dplyr::mutate(chg_per_cycle = tot_pct_cov - dplyr::lag(tot_pct_cov, order_by = Cycle))
+
+  }
+
+  return(Nat_Cov)
+}
+
+
+
+#' Native V. Non-native Cover Plot for Understory data
+#'
+#' @param data An Understory tibble/dataframe with Native and Non-native Cover
+#'
+#' @return Nat-v-Non Plot comparing Native and Non-native Cover (or Change in Cover)
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'
+#' Native_v_Nonnative_Plot <- UnderNativityCoverTotal.plot.(UnderNativityCoverTotal())
+#' }
+
+UnderNatCovTotal <- function(data, sample_cycle) {
```

Comment:
Like you mentioned, you can call UnderNativityCoverTotal from within this function instead of passing it in as data. Only other thing I'd change is putting "Plot" or "Graph" somewhere in the name of this function (and do the same for other functions that return plots).